rules_python failing with Bazel@HEAD in Bazel Downstream pipeline

https://buildkite.com/bazel/bazel-at-head-plus-downstream/builds/2675#0183d0a7-84bd-4541-b543-01d162c313ed

```
======================================================================
FAIL: test_match_toolchain (__main__.TestPythonVersion)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/var/lib/buildkite-agent/.cache/bazel/_bazel_buildkite-agent/eb882ad14e6111536f596011b6d53a9e/sandbox/linux-sandbox/48/execroot/rules_python/bazel-out/k8-fastbuild/bin/python/tests/toolchains/run_test_3.8.13.sh.runfiles/rules_python/python/tests/toolchains/3.8.13/run_acceptance_test.py", line 52, in test_match_toolchain
    self.assertEqual(output, "Python 3.8.13")
AssertionError: '' != 'Python 3.8.13'
+ Python 3.8.13
```

```
==================== Test output for //:test_vendored:
4c4
< from //:requirements.txt
---
> from @//:requirements.txt
FAIL: files "requirements.bzl" and "requirements.clean.bzl" differ. Please run:  bazel run //:vendor_requirements
```

Possible culprits are: https://github.com/bazelbuild/rules_python/pull/855 and https://github.com/bazelbuild/rules_python/pull/851

Ok, turns out its not, insofar as the CI is concerned:
 * https://buildkite.com/bazel/bazel-at-head-plus-disabled/builds/1550#0185dd3c-a7a5-48cc-b53d-e479e7b0ec18
 * https://storage.googleapis.com/bazel-untrusted-buildkite-artifacts/0185dd3c-a7a5-48cc-b53d-e479e7b0ec18/python/tests/toolchains/python_3_8_13_x86_64-unknown-linux-gnu_test/test_attempts/attempt_1.log

```
exec ${PAGER:-/usr/bin/less} "$0" || exit 1
Executing tests from //python/tests/toolchains:python_3_8_13_x86_64-unknown-linux-gnu_test
-----------------------------------------------------------------------------
2023/01/23 06:08:24 could not link local Bazel: could not copy file from /tmp/tmp6e5y4k72/bazel to /var/lib/buildkite-agent/.cache/bazelisk/local/-tmp-tmp6e5y4k72-bazel/bin/bazel: open /tmp/tmp6e5y4k72/bazel: no such file or directory
F/var/lib/buildkite-agent/.cache/bazel/_bazel_buildkite-agent/0bbb1d249f832c379464d3efbbd58cab/execroot/rules_python/external/python_3_11_1_x86_64-unknown-linux-gnu/lib/python3.11/subprocess.py:1125: ResourceWarning: subprocess 18 is still running
  _warn("subprocess %s is still running" % self.pid,
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/var/lib/buildkite-agent/.cache/bazel/_bazel_buildkite-agent/0bbb1d249f832c379464d3efbbd58cab/execroot/rules_python/external/python_3_11_1_x86_64-unknown-linux-gnu/lib/python3.11/unittest/case.py:622: ResourceWarning: unclosed file <_io.TextIOWrapper name=3 encoding='utf-8'>
  with outcome.testPartExecutor(self):
ResourceWarning: Enable tracemalloc to get the object allocation traceback

======================================================================
FAIL: test_match_toolchain (__main__.TestPythonVersion.test_match_toolchain)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/var/lib/buildkite-agent/.cache/bazel/_bazel_buildkite-agent/0bbb1d249f832c379464d3efbbd58cab/sandbox/linux-sandbox/48/execroot/rules_python/bazel-out/k8-fastbuild/bin/python/tests/toolchains/run_test_3.8.13.sh.runfiles/rules_python/python/tests/toolchains/3.8.13/run_acceptance_test.py", line 52, in test_match_toolchain
    self.assertEqual(output, "Python 3.8.13")
AssertionError: '' != 'Python 3.8.13'
+ Python 3.8.13

----------------------------------------------------------------------
Ran 1 test in 0.084s

FAILED (failures=1)
```

The only clue in there is something about not being able to copy bazel from one temp location to another? I'll have to dig into it a bit.

--- a/python/tests/toolchains/run_acceptance_test.py.tmpl
+++ b/python/tests/toolchains/run_acceptance_test.py.tmpl
@@ -23,9 +23,8 @@ class TestPythonVersion(unittest.TestCase):
         os.chdir("%test_location%")
         rules_python_path = os.path.join(os.environ["TEST_SRCDIR"], "rules_python")
 
+        test_tmpdir = os.environ["TEST_TMPDIR"]
         if %is_windows%:
-            test_tmpdir = os.environ["TEST_TMPDIR"]
-
             home = os.path.join(test_tmpdir, "HOME")
             os.mkdir(home)
             os.environ["HOME"] = home
@@ -33,6 +32,16 @@ class TestPythonVersion(unittest.TestCase):
             local_app_data = os.path.join(test_tmpdir, "LocalAppData")
             os.mkdir(local_app_data)
             os.environ["LocalAppData"] = local_app_data
+
+        # Bazelisk requires a cache directory be set
+        os.environ["XDG_CACHE_HOME"] = os.path.join(test_tmpdir, "xdg-cache-home")
+
+        # Unset this so this this works when called by Bazel's latest Bazel build
+        # pipeline. It sets the following combination, which interfer with each other:
+        # * --sandbox_tmpfs_path=/tmp
+        # * --test_env=USE_BAZEL_VERSION
+        # * USE_BAZEL_VERSION=/tmp/<something>
+        os.environ.pop("USE_BAZEL_VERSION", None)
 
         with open(".bazelrc", "w") as bazelrc:
             bazelrc.write(
@@ -47,8 +56,11 @@ class TestPythonVersion(unittest.TestCase):
             )
 
     def test_match_toolchain(self):
-        stream = os.popen("bazel run @python//:python3 -- --version")
-        output = stream.read().strip()
+        output = subprocess.check_output(
+          f"bazel run @python//:python3 -- --version",
+          shell = True, # Shell needed to look up via PATH
+          text=True,
+        ).strip()
         self.assertEqual(output, "Python %python_version%")
 
         subprocess.run("bazel test //...", shell=True, check=True)

--- a/python/tests/toolchains/run_acceptance_test.py.tmpl
+++ b/python/tests/toolchains/run_acceptance_test.py.tmpl
@@ -37,7 +37,7 @@ class TestPythonVersion(unittest.TestCase):
         os.environ["XDG_CACHE_HOME"] = os.path.join(test_tmpdir, "xdg-cache-home")
 
         # Unset this so this works when called by Bazel's latest Bazel build
-        # pipeline. It sets the following combination, which interfer with each other:
+        # pipeline. It sets the following combination, which interfere with each other:
         # * --sandbox_tmpfs_path=/tmp
         # * --test_env=USE_BAZEL_VERSION
         # * USE_BAZEL_VERSION=/tmp/<something>

--- a/python/tests/toolchains/run_acceptance_test.py.tmpl
+++ b/python/tests/toolchains/run_acceptance_test.py.tmpl
@@ -36,7 +36,7 @@ class TestPythonVersion(unittest.TestCase):
         # Bazelisk requires a cache directory be set
         os.environ["XDG_CACHE_HOME"] = os.path.join(test_tmpdir, "xdg-cache-home")
 
-        # Unset this so this this works when called by Bazel's latest Bazel build
+        # Unset this so this works when called by Bazel's latest Bazel build
         # pipeline. It sets the following combination, which interfer with each other:
         # * --sandbox_tmpfs_path=/tmp
         # * --test_env=USE_BAZEL_VERSION

